From 3bc79871ce2b8c6b020c54ac67d8bf1cb016d520 Mon Sep 17 00:00:00 2001
From: Charles Stevens <charles.stevens@compasses.com>
Date: Mon, 23 Nov 2020 17:11:13 -0600
Subject: [PATCH 32/44] RSOM-221 Spidev warning

Remove SPIDEV warning from code about using spidev compatible string directly.

We are using this string in out example DTS file(s) to allow userspace
access to "unknown" SPI devices connected to the RPi header.

Signed-off-by: Charles Stevens <charles.stevens@compasses.com>
---
 drivers/spi/spidev.c | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/drivers/spi/spidev.c b/drivers/spi/spidev.c
index e444e7cc6968..7b652bc59ced 100644
--- a/drivers/spi/spidev.c
+++ b/drivers/spi/spidev.c
@@ -727,15 +727,6 @@ static int spidev_probe(struct spi_device *spi)
 	int			status;
 	unsigned long		minor;
 
-	/*
-	 * spidev should never be referenced in DT without a specific
-	 * compatible string, it is a Linux implementation thing
-	 * rather than a description of the hardware.
-	 */
-	WARN(spi->dev.of_node &&
-	     of_device_is_compatible(spi->dev.of_node, "spidev"),
-	     "%pOF: buggy DT: spidev listed directly in DT\n", spi->dev.of_node);
-
 	spidev_probe_acpi(spi);
 
 	/* Allocate driver data */
-- 
2.17.1

